File: Algorithms 5/Lection 1/Question 6/main.cpp
#include <iostream>
#include <vector>

using namespace std;

string count_odd(vector<int>& nums) {
    int f = nums[0];
    bool is_odd = (f % 2 != 0);
    vector<char> result;

    for (size_t i = 1; i < nums.size(); ++i) {
        int s = nums[i];

        if (is_odd && s % 2 != 0) {
            result.push_back('x');
            f *= s;  // Оба нечётные → результат остаётся нечётным
        } else {
            result.push_back('+');
            f += s;
            is_odd = (f % 2 != 0);  // Обновляем чётность
        }
    }

    return string(result.begin(), result.end());
}

int main() {
    int n;
    cin >> n;

    vector<int> nums(n);
    for (int i = 0; i < n; ++i) {
        cin >> nums[i];
    }

    cout << count_odd(nums) << endl;
    return 0;
}

File: Algorithms 5/Lection 1/Question 6/main.py
"""
Этот код на питоне прошёл больше всего (11) тестов
Код на С++ проходит все тесты
Аналогичный код с С++ на пайтон проходит только 10 тестов
"""


from collections import deque


def count_odd(nums: deque[int]) -> str:
    f: int = nums.popleft()
    is_odd: bool = (f % 2 != 0)
    s: int = nums.popleft()
    result: deque[str] = deque()

    while nums or s is not None:
        if is_odd and s % 2 != 0:
            result.append('x')
            f *= s
        else:
            result.append('+')
            f += s
            if not is_odd and s % 2 == 0:
                is_odd = False
            else:
                is_odd = True
        s = nums.popleft() if nums else None
    return ''.join(result)


def main() -> None:
    n: int = int(input())
    nums: deque[int] = deque(map(int, input().split()))
    print(count_odd(nums))


if __name__ == "__main__":
    main()
